features/steps/unity_steps.rb

```diff
@@ -300,3 +300,81 @@ def spans_from_request_list(list)
       .flat_map { |s| s['spans'] }
       .select { |s| !s.nil? }
 end
+
+#
+# Find a span by name and return it
+#
+def find_span_by_name(span_name)
+  spans = spans_from_request_list(Maze::Server.list_for("traces"))
+  span = spans.find { |s| s['name'] == span_name }
+  raise Test::Unit::AssertionFailedError, "No span found with the name #{span_name}" if span.nil?
+  span
+end
+
+#
+# Find an attribute on a span by key
+#
+def find_span_attr(span, key)
+  attr = span['attributes']&.find { |a| a['key'] == key }
+  raise Test::Unit::AssertionFailedError, "No attribute '#{key}' found on span '#{span['name']}'" if attr.nil?
+  attr
+end
+
+Then('the span named {string} has an attribute {string}') do |span_name, attr_key|
```

Comment:
All the steps here are good candidates for adding to Maze Runner, but we already have steps inconsistently named across several repos - so for now I'll just add them to my spreadsheet once the PR is finalised and I'll consider them all in a big refactor.